tests/integration/test_charm_with_profile.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+import glob
+import time
+from pathlib import Path
+
+import lightkube
+import pytest
+import yaml
+from lightkube import codecs
+from lightkube.generic_resource import create_global_resource
+from pytest_operator.plugin import OpsTest
+
+basedir = Path("./").absolute()
+PROFILE_NAMESPACE = "profile-example"
+PROFILE_FILE_PATH = f"{basedir}/tests/integration/profile.yaml"
+PROFILE_FILE = yaml.safe_load(Path(PROFILE_FILE_PATH).read_text())
```

Comment:
`basedir` is already a path, so you can just append strings to it and have the result still be path. Then you can update a couple of other places in the code that convert `PROFILE_FILE_PATH` to a `Path` all over again, and edit `apply_manifests` accordingly (suggestions in corresponding comments).
```suggestion
PROFILE_FILE_PATH = basedir / "tests/integration/profile.yaml"
PROFILE_FILE = yaml.safe_load(PROFILE_FILE_PATH.read_text())
```